python/hsfs/core/arrow_flight_client.py

```diff
@@ -444,9 +444,7 @@ def create_training_dataset(
         self, feature_view_obj, training_dataset_obj, query_obj, arrow_flight_config
     ):
         training_dataset = {}
-        training_dataset["fs_name"] = util.strip_feature_store_suffix(
-            training_dataset_obj.feature_store_name
-        )
+        training_dataset["project_name"] = self._client._project_name
```

Comment:
I think it's fine here, because I don't think it's possible to create a training dataset in a shared feature store. 
But as a general note, be careful when you do this type of things `self._client._project_name` refers to the project the user is connected to, not the feature store they are working in.

So for example, if you do:
```
project = hopsworks.login(project="source_project")
fs = project.get_feature_store(name="shared_feature_store")
fv = fs.get_feature_view(...)
fv.create_training_dataset()
```

then your code above will be using `source_project` and not `shared_feature_store` project.